src/view/LoginView.java

```diff
@@ -134,7 +135,9 @@ public void actionPerformed(ActionEvent evt) {
     @Override
     public void propertyChange(PropertyChangeEvent evt) {
         LoginState state = (LoginState) evt.getNewValue();
-        setFields(state);
+        if (state.getUsernameError() != null) {
+            JOptionPane.showMessageDialog(this, state.getUsernameError());
```

Comment:
SUGGESTION: either add a comment here, or delete the obsolete passwordError attribute in the state and rename userError to simply error (or something similar).